eth/gasprice/gasprice.go

```diff
@@ -167,19 +409,89 @@ func (gpo *Oracle) getBlockPrices(ctx context.Context, signer types.Signer, bloc
 		return
 	}
 
+	blockTxs := block.Transactions()
+	txs := new(transactionsByGasPrice)
+	txs.txs = make([]*types.Transaction, len(blockTxs))
+	copy(txs.txs, blockTxs)
+	txs.baseFee = block.BaseFee()
+	sort.Sort(txs)
+
+	for _, tx := range txs.txs {
+		sender, err := types.Sender(signer, tx)
+		if err == nil && sender != block.Coinbase() {
+			price := tx.GasPrice()
+			if price == nil {
+				price = new(big.Int).Add(block.BaseFee(), tx.GasPremium())
+				if price.Cmp(tx.FeeCap()) > 0 {
+					price.Set(tx.FeeCap())
+				}
+			}
+			ch <- getBlockPricesResult{price, nil}
+			return
+		}
```

Comment:
```suggestion
		if err == nil && sender != block.Coinbase() {
		        continue
		}
		price := tx.GasPrice()
		if price == nil {
			price = new(big.Int).Add(block.BaseFee(), tx.GasPremium())
			if price.Cmp(tx.FeeCap()) > 0 {
				price.Set(tx.FeeCap())
			}
		}
		ch <- getBlockPricesResult{price, nil}
		return
```
To reduce the amount of nested if statements.